src/Adapter/Cart/QueryHandler/GetCartForOrderCreationHandler.php

```diff
@@ -278,6 +290,23 @@ private function extractProductsFromLegacySummary(Cart $cart, array $legacySumma
         return $products;
     }
 
+    /**
+     * @param Cart $cart
+     * @param array $legacySummary
+     * @param Currency $currency
+     *
+     * @return CartProduct[]
+     */
+    private function extractProductsFromLegacySummary(Cart $cart, array $legacySummary, Currency $currency): array
+    {
+        $products = [];
+        foreach ($legacySummary['products'] as &$product) {
```

Comment:
No need to use references here
```suggestion
        foreach ($legacySummary['products'] as $product) {
```